run-tests.sh

```diff
@@ -1,53 +1,56 @@
 #!/usr/bin/env bash
 
-# NOTE: set TEST_SUITE to be markers you want to run.
+# Tested on debian and alpine
+# You will need to install some dependencies yourself.
 
 set -eo pipefail
 
 export PYTHONIOENCODING="utf-8"
 export LANG=C.UTF-8
-export PIP_PROCESS_DEPENDENCY_LINKS="1"
+
 # Let's use a temporary cache directory
 export PIPENV_CACHE_DIR=`mktemp -d 2>/dev/null || mktemp -d -t 'pipenv_cache'`
 
-prefix() {
-	sed "s/^/   $1:    /"
-}
+# on some Linux OS python is python3
+PYTHON=${PYTHON:-"python"}
+PIPENV_PYTHON="${PIPENV_PYTHON:-3.7}"
 
-if [[ ! -z "$TEST_SUITE" ]]; then
-	echo "Using TEST_SUITE=$TEST_SUITE"
-fi
+PIP_CALL="${PIP_CALL:-${PYTHON} -m pip install --user}"
 
 HOME=$(readlink -f ~/)
+
 if [[ -z "$HOME" ]]; then
     if [[ "$USER" == "root" ]]; then
         HOME="/root"
     fi
 fi
+
 if [[ ! -z "$HOME" ]]; then
     export PATH="${HOME}/.local/bin:${PATH}"
 fi
-pip install certifi
-export GIT_SSL_CAINFO=$(python -m certifi)
-echo "Path: $PATH"
-echo "Installing Pipenv..."
-python -m pip install --upgrade -e "$(pwd)" setuptools wheel pip
-VENV_CMD="python -m pipenv --venv"
+
+# This installs the dependencies for pipenv
+${PIP_CALL} --upgrade pip setuptools wheel virtualenv --upgrade-strategy=eager
+
+VENV_CMD="${PYTHON} -m pipenv --venv"
 RM_CMD="pipenv --rm"
-echo "$ PIPENV_PYTHON=2.7 $VENV_CMD && PIPENV_PYTHON=2.7 $RM_CMD"
-echo "$ PIPENV_PYTHON=3.7 $VENV_CMD && PIPENV_PYTHON=3.7 $RM_CMD"
-{ PIPENV_PYTHON=2.7 $VENV_CMD && PIPENV_PYTHON=2.7 $RM_CMD ; PIPENV_PYTHON=3.7 $VENV_CMD && PIPENV_PYTHON=3.7 $RM_CMD ; }
+
+echo "$ PIPENV_PYTHON=${PIPENV_PYTHON} $VENV_CMD && PIPENV_PYTHON=${PIPENV_PYTHON} $RM_CMD"
```

Comment:
Maybe this could be made a variable so that it isn't repeated between the `echo` and the actual execution of it on line 40 below?

Reply:
Can you clarify here? I don't understand your comment.

Reply:
Just was noticing that echo statement is a repeated construction of the command below it, maybe it could be made a variable?   Not a blocker and I am not a bash scripting expert, so might not be worth it.

Reply:
TBH, These `echo` statements are quite obsolete. If one wants to see what the script does,
there is always `bash -x run-tests.sh`.
I'd leave it like these. In the near future I am going to refactor the makefile, so there is no point
investing too much time here.